### PR TITLE
feat: Allow configuring cluster_autoscaling for safer cluster variants

### PR DESCRIPTION
**Pull request by samschlegel**
Does what it says on the tin 😄

### CODE REVIEW

File: autogen/safer-cluster/variables.tf.tmpl [hunk 1]
```diff
@@ -202,6 +202,28 @@ variable "node_pools_oauth_scopes" {
...
+  type = object({
+    enabled             = bool
+    autoscaling_profile = string
+    min_cpu_cores       = number
+    max_cpu_cores       = number
+    min_memory_gb       = number
+    max_memory_gb       = number
+    gpu_resources       = list(object({ resource_type = string, minimum = number, maximum = number }))
+  })
+  default = {
```
Comment (samschlegel) [inline autogen/safer-cluster/variables.tf.tmpl:215]: I don't love that this is duplicated from the underlying beta cluster, but making it nullable and defaulting to null feels like it hides defaults too much

**Comment by samschlegel** [comment]
I also couldn't get `make build` running on my local because of some strange Docker error that I don't particularly care to hack around if possible :joy: https://github.com/moby/moby/issues/43576

**Comment by comment-bot-dev** [comment]
@samschlegel
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_generate_modules
 The modules need to be regenerated. Please run `make_build`. 
 ```bash 
Checking submodule's files generation
diff -r '--exclude=.terraform' '--exclude=.kitchen' '--exclude=.git' /workspace/modules/safer-cluster/README.md /tmp/tmp.ViPMr6rH4P/workspace/modules/safer-cluster/README.md
206a207
> | cluster\_autoscaling | Cluster autoscaling configuration. See [more details](https://cloud.google.com/kubernetes-engine/docs/reference/rest/v1beta1/projects.locations.clusters#clusterautoscaling) | <pre>object({<br>    enabled             = bool<br>    autoscaling_profile = string<br>    min_cpu_cores       = number<br>    max_cpu_cores       = number<br>    min_memory_gb       = number<br>    max_memory_gb       = number<br>    gpu_resources       = list(object({ resource_type = string, minimum = number, maximum = number }))<br>  })</pre> | <pre>{<br>  "autoscaling_profile": "BALANCED",<br>  "enabled": false,<br>  "gpu_resources": [],<br>  "max_cpu_cores": 0,<br>  "max_memory_gb": 0,<br>  "min_cpu_cores": 0,<br>  "min_memory_gb": 0<br>}</pre> | no |
diff -r '--exclude=.terraform' '--exclude=.kitchen' '--exclude=.git' /workspace/modules/safer-cluster-update-variant/README.md /tmp/tmp.ViPMr6rH4P/workspace/modules/safer-cluster-update-variant/README.md
206a207
> | cluster\_autoscaling | Cluster autoscaling configuration. See [more details](https://cloud.google.com/kubernetes-engine/docs/reference/rest/v1beta1/projects.locations.clusters#clusterautoscaling) | <pre>object({<br>    enabled             = bool<br>    autoscaling_profile = string<br>    min_cpu_cores       = number<br>    max_cpu_cores       = number<br>    min_memory_gb       = number<br>    max_memory_gb       = number<br>    gpu_resources       = list(object({ resource_type = string, minimum = number, maximum = number }))<br>  })</pre> | <pre>{<br>  "autoscaling_profile": "BALANCED",<br>  "enabled": false,<br>  "gpu_resources": [],<br>  "max_cpu_cores": 0,<br>  "max_memory_gb": 0,<br>  "min_cpu_cores": 0,<br>  "min_memory_gb": 0<br>}</pre> | no |
diff -r '--exclude=.terraform' '--exclude=.kitchen' '--exclude=.git' /workspace/modules/safer-cluster-update-variant/variables.tf /tmp/tmp.ViPMr6rH4P/workspace/modules/safer-cluster-update-variant/variables.tf
144,165d143
< variable "cluster_autoscaling" {
<   type = object({
<     enabled             = bool
<     autoscaling_profile = string
<     min_cpu_cores       = number
<     max_cpu_cores       = number
<     min_memory_gb       = number
<     max_memory_gb       = number
<     gpu_resources       = list(object({ resource_type = string, minimum = number, maximum = number }))
<   })
<   default = {
<     enabled             = false
<     autoscaling_profile = "BALANCED"
<     max_cpu_cores       = 0
<     min_cpu_cores       = 0
<     max_memory_gb       = 0
<     min_memory_gb       = 0
<     gpu_resources       = []
<   }
<   description = "Cluster autoscaling configuration. See [more details](https://cloud.google.com/kubernetes-engine/docs/reference/rest/v1beta1/projects.locations.clusters#clusterautoscaling)"
< }
< 
224a203,224
> }
> 
> variable "cluster_autoscaling" {
>   type = object({
>     enabled             = bool
>     autoscaling_profile = string
>     min_cpu_cores       = number
>     max_cpu_cores       = number
>     min_memory_gb       = number
>     max_memory_gb       = number
>     gpu_resources       = list(object({ resource_type = string, minimum = number, maximum = number }))
>   })
>   default = {
>     enabled             = false
>     autoscaling_profile = "BALANCED"
>     max_cpu_cores       = 0
>     min_cpu_cores       = 0
>     max_memory_gb       = 0
>     min_memory_gb       = 0
>     gpu_resources       = []
>   }
>   description = "Cluster autoscaling configuration. See [more details](https://cloud.google.com/kubernetes-engine/docs/reference/rest/v1beta1/projects.locations.clusters#clusterautoscaling)"
Error: submodule's files generation has not been run, please run the
'make build' command and commit changes
 ``` 
- ⚠️check_documentation
 The documentation needs to be regenerated. Please run `make generate_docs`. 
 ```bash 
Checking for documentation generation
diff -r '--exclude=.terraform' '--exclude=.kitchen' '--exclude=.git' '--exclude=autogen' '--exclude=*.tfvars' /workspace/modules/safer-cluster/README.md /tmp/tmp.1vJC5rI4Gb/generate_docs/workspace/modules/safer-cluster/README.md
206a207
> | cluster\_autoscaling | Cluster autoscaling configuration. See [more details](https://cloud.google.com/kubernetes-engine/docs/reference/rest/v1beta1/projects.locations.clusters#clusterautoscaling) | <pre>object({<br>    enabled             = bool<br>    autoscaling_profile = string<br>    min_cpu_cores       = number<br>    max_cpu_cores       = number<br>    min_memory_gb       = number<br>    max_memory_gb       = number<br>    gpu_resources       = list(object({ resource_type = string, minimum = number, maximum = number }))<br>  })</pre> | <pre>{<br>  "autoscaling_profile": "BALANCED",<br>  "enabled": false,<br>  "gpu_resources": [],<br>  "max_cpu_cores": 0,<br>  "max_memory_gb": 0,<br>  "min_cpu_cores": 0,<br>  "min_memory_gb": 0<br>}</pre> | no |
diff -r '--exclude=.terraform' '--exclude=.kitchen' '--exclude=.git' '--exclude=autogen' '--exclude=*.tfvars' /workspace/modules/safer-cluster-update-variant/README.md /tmp/tmp.1vJC5rI4Gb/generate_docs/workspace/modules/safer-cluster-update-variant/README.md
206a207
> | cluster\_autoscaling | Cluster autoscaling configuration. See [more details](https://cloud.google.com/kubernetes-engine/docs/reference/rest/v1beta1/projects.locations.clusters#clusterautoscaling) | <pre>object({<br>    enabled             = bool<br>    autoscaling_profile = string<br>    min_cpu_cores       = number<br>    max_cpu_cores       = number<br>    min_memory_gb       = number<br>    max_memory_gb       = number<br>    gpu_resources       = list(object({ resource_type = string, minimum = number, maximum = number }))<br>  })</pre> | <pre>{<br>  "autoscaling_profile": "BALANCED",<br>  "enabled": false,<br>  "gpu_resources": [],<br>  "max_cpu_cores": 0,<br>  "max_memory_gb": 0,<br>  "min_cpu_cores": 0,<br>  "min_memory_gb": 0<br>}</pre> | no |
Error: Documentation generation has not been run, please run the
'make docker_generate_docs' command and commit the above changes.
 ```

**Comment by comment-bot-dev** [comment]
@samschlegel
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Review by bharathkkb — APPROVED** [review]
Thanks for the PR @samschlegel

File: autogen/safer-cluster/variables.tf.tmpl (hunk 1 above)
Comment (bharathkkb) [reply autogen/safer-cluster/variables.tf.tmpl:215]: Yeah we may look at rearchitecting some of these objects with optionals in the future now that 1.3 supports it.

**Comment by jawnsy** [comment]
@samschlegel FYI in the referenced bug, looks like it's an issue with docker running in rootless mode -- I've run into the same thing when running docker under sysbox, which is also using user namespacing to run rootlessly.

I was able to run all the tools normally using podman in the default rootless mode (I'm running on macOS, so it's inside a VM)

https://github.com/moby/moby/issues/43576#issuecomment-1136056638

**Pull request merged**